### PR TITLE
[r] Fix creation of SOMADataFrames with non-int32 dimensions

### PR DESCRIPTION
**Pull request by aaronwolen**
Minor tweaks to `SOMADataFrame` to support creation of dimensions with int64, float/double, and string data types. As per the spec, we limit the domain of numeric dimensions to non-negative values.

### CODE REVIEW

File: apis/r/R/SOMADataFrame.R [hunk 1]
```diff
@@ -38,11 +38,21 @@ SOMADataFrame <- R6::R6Class(
       for (field_name in index_column_names) {
         field <- schema$GetFieldByName(field_name)
 
+        # TODO: Parameterize
+        tile_extent <- 2048L
```
Comment (johnkerl) [inline apis/r/R/SOMADataFrame.R:42]: recent tiledb-soma-py bug: if index type is int8/uint8 this mods to 0 😬 -- maybe use 64 in that case

**Review by johnkerl — APPROVED** [review]

**Review by eddelbuettel — APPROVED** [review]
Looks good

File: apis/r/R/utils-arrow.R [hunk 2]
```diff
@@ -112,18 +112,35 @@ arrow_type_range <- function(x) {
     uint32 = bit64::as.integer64(c(0, 4294967295)),
     # We can't specify the full range of uint64 in R so we use the max of int64
     uint64 = c(bit64::as.integer64(0), bit64::lim.integer64()[2]),
-    float32 = c(-3.4028235e+38, 3.4028235e+38),
+    # float32/float
     float = c(-3.4028235e+38, 3.4028235e+38),
-    float64 = c(-1.7976931348623157e+308, 1.7976931348623157e+308),
+    # float64/double
     double =  c(-1.7976931348623157e+308, 1.7976931348623157e+308),
```
Comment (eddelbuettel) [inline apis/r/R/utils-arrow.R:118]: If you want to be fancy you could use R constants here:

```r
> c(.Machine$double.xmin, .Machine$double.xmax)
[1] 2.22507e-308 1.79769e+308
> 
```

Comment (aaronwolen) [reply apis/r/R/utils-arrow.R:118]: I _always_ want to be fancy 😄. Great suggestion!

File: apis/r/R/SOMADataFrame.R (hunk 1 above)
Comment (aaronwolen) [reply apis/r/R/SOMADataFrame.R:42]: Good call, thanks!

File: apis/r/R/utils-arrow.R (hunk 2 above)
Comment (aaronwolen) [reply apis/r/R/utils-arrow.R]: Fixed in d20770aa and tested in 45b1a7ad

**Pull request merged**